### ISSUE TITLE
browser connections have not been established:

### ISSUE THREAD

**Issue by chintsan** [DevExpress/testcafe#7502]
### What is your Scenario?

Hi, 

we have our end to end tests setup using testcafe 1.19.0. Tests were running fine on Jenkins till this Monday eod but suddenly we are facing issue from that point till time.  Please refer attached error and appreciate your input on this.

Surprisingly same suite is passing in local. 

We did below to mitigate this but no luck.

- increase browserinittimeout
- ran with --no-sandbox flag

<img width="1689" alt="Screen Shot 2023-02-08 at 4 46 30 pm" src="https://user-images.githubusercontent.com/15031528/217445300-da508dab-b3e6-40cc-9d96-28f71f03c049.png">

 

### What is the Current behavior?

tests are failing as browser doesn't pop up.

### What is the Expected behavior?

Browser should be invoked for test cafe to conduct tests.

### What is your public website URL? (or attach your complete example)

N/A

### What is your TestCafe test code?

createTestCafe('localhost')
  .then((tc) => {
    testcafe = tc;
    const runner = testcafe.createRunner();

    return runner
      .compilerOptions({
        typescript: {
          options: {
            typeRoots: ['../../node_modules/@types', '../../typings'],
          },
        },
      })
      .src(runnerConfig.paths)
      .browsers([usedBrowser])
      .concurrency(runnerConfig.concurrency)
      .screenshots({
        path: `screenshots/${usedTestType}`,
        takeOnFails: true,
        fullPage: true,
      })
      .reporter([
        'list',
        {
          name: 'xunit',
          output: `report/unit/testcafe/e2e/res${_.capitalize(
            browser
          )}_${_.capitalize(
            usedTestType === 'normal' ? '' : usedTestType
          )}.xml`,
        },
      ])
      .run({
        skipJsErrors: true,
        quarantineMode: runnerConfig.quarantineMode,
        selectorTimeout: 20000,
        assertionTimeout: 10000,
        browserInitTimeout: 600000,
        pageLoadTimeout: 0,
        stopOnFirstFail,
        speed: 1,
      });
  })

### Steps to Reproduce

1.No specific steps. tests fail on CI infrastructure / in jenkins pipeline 
2.
3.

### TestCafe version

1.19.0

### Node.js version

16.19.0

### Command-line arguments

 npm run test:e2e:chrome:headless:jenkins "--" "--browser" "chromium" "--testType" "test"

**Comment by chintsan** [comment]
Hi, upgraded to testcafe 2.3.0 but unfortunately issue persists.

<img width="539" alt="Screen Shot 2023-02-08 at 6 38 47 pm" src="https://user-images.githubusercontent.com/15031528/217464759-3ac194c3-45fe-418b-9d6b-2fb8bb24ffd0.png">

**Comment by AlexKamaev** [comment]
Unfortunately, this information is insufficient to find the cause of the issue. From your configuration file, I can only assume that there can be a problem with the `concurrency` option. Please try to reduce the number of parallel browsers or temporarily remove this option.
Any additional research requires a full project, a set of tests, the `package.json` file, and instructions on how to start your test on the Jenskins.

**Comment by chintsan** [comment]
Hi, 
concurrency is one.

**Comment by chintsan** [comment]
Hi, fyi this issue isn't resolve and i didn't closed this issue as against what is shown above.

**Comment by AlexKamaev** [comment]
It's not quite clear why the issue happened. If you say that the concurrency was the cause, this can mean that the settings of your CI have changed somehow.
Unfortunately, we cannot do much without a reproducible example.

**Comment by chintsan** [comment]
I think we were not suspecting concurrency  though its value was clarified upon a query in this thread.

What we know as of now is that our Jenkins is configured to pull latest testcafe image from docker, Setup container etc and then run e2e tests.

 we are following same process and there hasn't been a change in this but issue started on Monday, 06 Feb 2023. Jenkins continues to pull and run with latest testcafe image where as local uses a lower version (1.19.0 and tested on 2.3.0) and completes tests successfully.

If you can confirm that browser connection isn't a issue in any of the recent releases that would help.

**Comment by AlexKamaev** [comment]
We did not get any questions related to the connection establishing last weeks. We did not add any changes in the browser opening mechanism either. At present the cause of the issue is still unclear for us, that's why we asked you to share the details mentioned in the [https://github.com/DevExpress/testcafe/issues/7502#issuecomment-1423769356](https://github.com/DevExpress/testcafe/issues/7502#issuecomment-1423769356) comment.
